version: '3.9'

services:
  capy-core-database:
    image: postgres:16
    container_name: capy-core-database
    restart: always
    ports:
      - '6968:5432'
    env_file:
      - ./backend/core/.env
    volumes:
      - pg_data:/var/lib/postgresql/data
    networks:
      - capy-world-network

  capy-core-server:
    build:
      context: ./backend/core
      dockerfile: ./Dockerfile
      target: dev
    container_name: capy_core_server
    command: sh -c "./entrypoint.sh"
    restart: always
    ports:
      - '6969:6969'
    env_file:
      - ./backend/core/.env
    volumes:
      - ./backend/core/src:/app/backend/src
      - ./backend/core/config:/app/backend/config
      - static-data:/app/backend/core/config/static
      - media-data:/app/backend/core/config/media
    networks:
      - capy-world-network

  capy-core-nginx:
    build:
      context: ./nginx
      dockerfile: ./Dockerfile
      target: dev
    container_name: nginx
    restart: always
    ports:
      - '80:8080'
    volumes:
      - static-data:/app/backend/core/config/static
      - media-data:/app/backend/core/config/media
    networks:
      - capy-world-network

  capy-core-rabbitmq:
    image: rabbitmq:3.13.1-management
    container_name: capy-core-rabbitmq
    ports:
      - "15627:15672"
      - "5627:5672"
    env_file:
      - ./backend/ws-server/.env
    volumes:
      - ./rabbitmq:/var/lib/rabbitmq
    networks:
      - capy-world-network
    healthcheck:
      test: [ "CMD", "rabbitmqctl", "status" ]
      interval: 10s
      timeout: 10s
      retries: 5

volumes:
  pg_data:
  static-data:
  media-data:

networks:
  capy-world-network:
    driver: bridge